src/packagedcode/plugin_package.py

```diff
@@ -272,7 +278,74 @@ def process_codebase(self, codebase, strip_root=False, package_only=False, **kwa
                 if TRACE_LICENSE and modified:
                     logger_debug(f'packagedcode: process_codebase: add_referenced_license_matches_from_package: modified: {modified}')
 
+@post_scan_impl
+class PackageSummary(PostScanPlugin):
+    """
+    Summary at the Package Level.
+    """
+    run_order = 11
+    sort_order= 11
+
+    options = [
+        PluggableCommandLineOption(('--package-summary',),
+        is_flag=True, default=False,
+        help='Summarize scans by providing License Clarity Score' 
+        'and populating other license/copyright attributes '
+        'for package instances.',
+        required_options=['classify'],
+        help_group=POST_SCAN_GROUP)
+    ]
+
+    def is_enabled(self, package_summary, **kwargs):
+        return package_summary
 
+    def process_codebase(self, codebase, package_summary, **kwargs):
+        """
+        Process the codebase.
+        """
+        if not self.is_enabled(package_summary):
+            return
+        
+        packages = codebase.attributes.packages
+        resource_for_packages = list(get_files_for_packages(codebase))
+        packages_copy = copy.deepcopy(packages) # created a deep copy 
+        package_resources = {}
+        package_attributes_map={}
+        for resource, package_uid in resource_for_packages:
```

Comment:
This part should be in a different function, utilising `get_files_for_packages` and returning `package_resources`

Reply:
Resolved